class.jetpack-jitm.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+/**
+ * Jetpack just in time messaging through out the admin
+ *
+ */
+class Jetpack_JITM {
+
+	/**
+	 * @var Jetpack_JITM
+	 **/
+	private static $instance = null;
+
+	static function init() {
+		if ( is_null( self::$instance ) ) {
+			self::$instance = new Jetpack_JITM;
+		}
+		return self::$instance;
+	}
+
+	private function __construct() {
+		add_action( 'install_plugins_search', array( $this, 'plugin_search' ) );
+		add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_css_header' ) );
+	}
+
+	/*
+	 * Present JITM on plugin search results page
+	 * so we can better educate users on what Jetpack does already
+	 *
+	 */
+	function plugin_search() {
+		$hide_jitm = Jetpack_Options::get_option( 'hide_jitm_plugins' );
+
+		if ( $hide_jitm ) {
+			return;
+		}
+
+		$search_term = strtolower( $_GET['s'] );
+
+		$module_info = $this->jitm_module_tags( $search_term );
+
+		//If we have a JITM to show, show it in a notice above the search results
+		if ( $module_info ){
+
+			//display content
+			echo '<div class="jp-jitm"><a href="#" class="dismiss"><span class="genericon genericon-close"></span></a><p><span class="icon"></span>';
+			_e( 'Jetpack is already here to help. Click here to learn more about ', 'jetpack' );
+			echo '<a href="' . Jetpack::admin_url() . '_modules&info=' . $module_info[0]['module_slug'] .'" class="jetpack-learnmore-module" id="jetpack-'.$module_info[0]['module_slug'].'">';
```

Comment:
Would it be possible to include the module name in the string with `printf`? It would make it easier for translators, as in some languages the module name won't be at the end of the sentence.